pylhc/machine_settings_info.py

```diff
@@ -198,13 +210,21 @@ def get_info(opt) -> Dict[str, object]:
     except ValueError as e:
         LOG.error(str(e))
     else:
-        trim_histories = LSA.get_trim_history(beamprocess_info.Object, opt.knobs, start_time=acc_start_time, end_time=acc_time, accelerator=opt.accel)
-        trims = _get_last_trim(trim_histories)
+        if opt.knobs is not None:
+            if len(opt.knobs) == 1 and opt.knobs[0].lower() == 'all':
+                opt.knobs = []  # will extract all knobs in get_trim_history
+            if len(opt.knobs) == 1 and opt.knobs[0].lower() == 'default':
```

Comment:
`elif` maybe, not a big deal

Reply:
yeah. had it first, but code looks nicer this way (I think we can accept the inefficiency)